src/containers/DAO/pages/DAOPage/Proposals/CreateProposal/hooks/useCreateProposal.ts

```diff
@@ -0,0 +1,89 @@
+import { useState, useMemo, useCallback } from 'react';
+import { useHistory } from 'react-router-dom';
+import { useWeb3React } from '@web3-react/core';
+import { Web3Provider } from '@ethersproject/providers/lib/web3-provider';
+import type { zDAO } from '@zero-tech/zdao-sdk';
+import { useDidMount } from 'lib/hooks/useDidMount';
+import { useWillUnmount } from 'lib/hooks/useWillUnmount';
+import { getTokenOptionsFromAssets } from '../CreateProposal.helpers';
+import { DAO_CREATE_PROPPAL } from '../../../Proposals/Proposals.constants';
+import useAssets from '../../../hooks/useAssets';
+
+export const useCreateProposal = (dao?: zDAO) => {
+	const [triggerCancel, setTriggerCancel] = useState<boolean>(false);
+	const [showWalletConnectModal, setShowWalletConnectModal] =
+		useState<boolean>(false);
+
+	// - Life Cycle
+	useDidMount(() => {
+		const nav = document.getElementById('dao-page-nav-tabs');
+		if (nav) {
+			nav.style.display = 'none';
+		}
+	});
+
+	useWillUnmount(() => {
+		const nav = document.getElementById('dao-page-nav-tabs');
+		if (nav) {
+			nav.style.display = 'block';
+		}
+	});
```

Comment:
nit: could be simplified

```js
useEffect(() => {
    const nav = document.getElementById('dao-page-nav-tabs');
    if(nav) {
        nav.style.display = 'none';
    }
   return () => {
        nav.style.display = 'block';
    }
}, [])
```

I think this functionality might be outside of the scope of this hook though, this hook is doing a lot of stuff.